onnxruntime/core/mlas/lib/qgemm.cpp

```diff
@@ -79,125 +77,266 @@ MlasGemmU8X8ScaleSumBuffer(
+            }
+            else {
+                ProcessImpl<true, MLAS_QGEMM_OUTPUT_MODE::ZeroMode, MLAS_QUANTIZATION_GRANULARITY::PerColumn>(
+                    C,
+                    StartM,
+                    StartN,
+                    CountM,
+                    CountN,
+                    ldc);
+            }
+        } else if(OutputMode_ == MLAS_QGEMM_OUTPUT_MODE::AccumulateMode){
+            ProcessImpl<true, MLAS_QGEMM_OUTPUT_MODE::AccumulateMode, MLAS_QUANTIZATION_GRANULARITY::PerMatrix>(
+                C,
+                StartM,
+                StartN,
+                CountM,
+                CountN,
+                ldc);
+        } else{
+            ProcessImpl<true, MLAS_QGEMM_OUTPUT_MODE::ZeroMode, MLAS_QUANTIZATION_GRANULARITY::PerMatrix>(
+                C,
+                StartM,
+                StartN,
+                CountM,
+                CountN,
+                ldc);
+        }
+    } else{
+        if (QuantGran_ == MLAS_QUANTIZATION_GRANULARITY::PerColumn) {
+            if (OutputMode_ == MLAS_QGEMM_OUTPUT_MODE::AccumulateMode) {
+                ProcessImpl<false, MLAS_QGEMM_OUTPUT_MODE::AccumulateMode, MLAS_QUANTIZATION_GRANULARITY::PerColumn>(
+                    C,
+                    StartM,
+                    StartN,
+                    CountM,
+                    CountN,
+                    ldc);
+            }
+            else {
+                ProcessImpl<false, MLAS_QGEMM_OUTPUT_MODE::ZeroMode, MLAS_QUANTIZATION_GRANULARITY::PerColumn>(
+                    C,
+                    StartM,
+                    StartN,
+                    CountM,
+                    CountN,
+                    ldc);
+            }
+        }
+        else if (OutputMode_ == MLAS_QGEMM_OUTPUT_MODE::AccumulateMode) {
+            ProcessImpl<false, MLAS_QGEMM_OUTPUT_MODE::AccumulateMode, MLAS_QUANTIZATION_GRANULARITY::PerMatrix>(
+                C,
+                StartM,
+                StartN,
+                CountM,
+                CountN,
+                ldc);
+        }
+        else {
+            ProcessImpl<false, MLAS_QGEMM_OUTPUT_MODE::ZeroMode, MLAS_QUANTIZATION_GRANULARITY::PerMatrix>(
+                C,
+                StartM,
+                StartN,
+                CountM,
+                CountN,
+                ldc);
+        }
+    }
+}
+
+template<bool HASBIAS, MLAS_QGEMM_OUTPUT_MODE MODE, MLAS_QUANTIZATION_GRANULARITY QUANTGRAN>
+inline
 void
-MlasGemmU8X8OutputFloat(
-    const MLAS_GEMM_U8X8_WORK_BLOCK* WorkBlock,
-    int32_t* C,
+MLAS_QGEMM_SCALE_BIAS_OUTPUT_PROCESSOR::ProcessImpl(
+    const int32_t* C,
+    size_t StartM,
     size_t StartN,
     size_t CountM,
-    size_t CountN
-    )
+    size_t CountN,
+    size_t ldc) const
 /*++
 
 Routine Description:
 
-    This routine converts the output matrix to a floating point format using
-    the supplied scale and bias parameters.
+    This routine process the output of quantized GEMM.
```

Comment:
Probably need a little more detail here. Maybe use the original comment but instead of "supplied" change to "stored"?